### PR TITLE
Selet AI Tests for Vector Index.

### PR DESCRIPTION
**Pull request by Pushpendra-Garg**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/credential/conftest.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        "cred_password": get_credential_env_value(
+            "CRED_PASSWORD", default_value="OCI credential password"
+        ),
+    }
+
+
+@pytest.fixture(scope="session")
+def credential_connect_as(test_env):
+    def _connect_as(admin=False, **overrides):
+        select_ai.disconnect()
```
Comment (aosingh) [inline tests/credential/conftest.py]: Please don't run select_ai.disconnect() in individual conftest. 

We have seen this causes issues as it modifies the global connection state. We already have connect/disconnect in the root level conftest.py which is automatically run for every module.  

https://github.com/oracle/python-select-ai/blob/main/tests/conftest.py#L201

We should not do any connection management in individual test suites.

File: tests/credential/conftest.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        connect_params.update(overrides)
+        select_ai.connect(**connect_params)
+
+    return _connect_as
+
+
+@pytest.fixture(scope="session")
+def credential_async_connect_as(test_env):
+    async def _connect_as(admin=False, **overrides):
+        await select_ai.async_disconnect()
```
Comment (aosingh) [inline tests/credential/conftest.py]: We already have async connect and disconnect here 
https://github.com/oracle/python-select-ai/blob/main/tests/conftest.py#L208

No need to connect here

**Comment by aosingh** [comment]
@Pushpendra-Garg please rebase changes from main. Thanks

**Comment by Pushpendra-Garg** [comment]
> @Pushpendra-Garg please rebase changes from main. Thanks

Done.

File: tests/credential/conftest.py (hunk 2 above)
Comment (Pushpendra-Garg) [reply tests/credential/conftest.py]: Removed additional connect/disconnect from subfolders.

File: tests/credential/conftest.py (hunk 1 above)
Comment (Pushpendra-Garg) [reply tests/credential/conftest.py]: Removed additional connect/disconnect from subfolders.

**Review by prateeks99 — DISMISSED** [review]

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -36,6 +36,49 @@ jobs:
           pip install pytest anyio
           pip install -e .
 
+      - name: Debug vector index secret wiring
```
Comment (aosingh) [inline .github/workflows/test.yaml]: Can we please remove this debug step ?

File: tests/credential/conftest.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2025, Oracle and/or its affiliates.
```
Comment (aosingh) [inline tests/credential/conftest.py]: For new files, Copyright header should use the current year - 2026

File: tests/credential/test_2200_async_create_cred.py [hunk 5]
```diff
@@ -0,0 +1,415 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2025, Oracle and/or its affiliates.
```
Comment (aosingh) [inline tests/credential/test_2200_async_create_cred.py]: 2026

File: tests/credential/test_2200_create_cred.py [hunk 6]
```diff
@@ -0,0 +1,356 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2025, Oracle and/or its affiliates.
```
Comment (aosingh) [inline tests/credential/test_2200_create_cred.py]: Please update the year in all files

File: tests/vector_index/test_5500_enable_disable_index.py [hunk 7]
```diff
@@ -0,0 +1,360 @@
...
+            select_ai.delete_credential("OBJSTORE_CRED", force=True)
+            logger.info("Deleted credential 'OBJSTORE_CRED'")
+        except Exception as e:
+            logger.warning(f"delete_credential() raised {e} unexpectedly.")
+
+    def setup_method(self, method):
+        logger.info(f"SetUp for {method.__name__}")
+        self.objstore_cred = "OBJSTORE_CRED"
+        self.vecidx = select_ai.VectorIndex()
+        self.vector_index = list(self.vecidx.list(index_name_pattern=".*"))[0]
```
Comment (aosingh) [inline tests/vector_index/test_5500_enable_disable_index.py]: This will operate on whichever vector index happens to be returned first, not the index created in setup. In a shared schema or after earlier tests, this can validate/disable the wrong object. Fetch by exact name, for example index_name_pattern=f"^{self.index_name}$", and assert exactly one match.

File: tests/credential/test_2200_create_cred.py [hunk 8]
```diff
@@ -0,0 +1,356 @@
...
+        try:
+            with pytest.raises(DatabaseNotConnectedError):
+                select_ai.create_credential(credential=credential, replace=False)
+            self.logger.info("Expected DatabaseNotConnectedError raised.")
+        finally:
+            select_ai.create_pool(**self.test_env.connect_params(use_pool=True))
+
+    def test_2217(self):
+        """Test Credential creation for a local test user"""
+        test_username = "TEST_USER1"
```
Comment (aosingh) [inline tests/credential/test_2200_create_cred.py]: We run tests for different Python versions in parallel. Creating TEST_USER1 is unsafe for parallel runs. Use UUID/suffix-based names derived from the test session, and always drop them in `finally`.

File: tests/provider/test_2400_async_enable.py [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+
+    cls = request.cls
+    cls.user = cls.provider_params["user"]
+    cls.password = cls.provider_params["password"]
+    cls.dsn = cls.provider_params["dsn"]
+    cls.db_users = []
+
+    try:
+        for i in range(1, 6):
+            user = f"DB_USER{i}"
```
Comment (aosingh) [inline tests/provider/test_2400_async_enable.py]: Don't use user names like DB_USER1, DB_USER2. In a shared CI infrastructure this can break parallel runs. Use UUID/suffix-based names derived from the test session, and always drop them in `finally`

File: tests/vector_index/test_5400_list_index.py [hunk 10]
```diff
@@ -0,0 +1,350 @@
...
+        logger.info("Invalid type pattern handled correctly with empty result.")
+
+    # ----------------
+    # Stress / Edge cases
+    # ----------------
+    def test_5413(self):
+        """Test None as pattern input."""
+        logger.info("Testing None as pattern input...")
+        indexes = self.vector_index.list(None)
+        assert len(list(indexes)) != len(self.indexes)
```
Comment (aosingh) [inline tests/vector_index/test_5400_list_index.py]: What are we testing here ?

File: tests/vector_index/test_5400_list_index.py [hunk 11]
```diff
@@ -0,0 +1,350 @@
...
+        logger.info("None pattern handled correctly.")
+
+    def test_5414(self):
+        """Test empty string as pattern input."""
+        logger.info("Testing empty string as pattern input...")
+        indexes = self.vector_index.list("")
+        assert len(list(indexes)) != len(self.indexes)
```
Comment (aosingh) [inline tests/vector_index/test_5400_list_index.py]: Same, what is tested here ?

File: .github/workflows/test.yaml (hunk 3 above)
Comment (Pushpendra-Garg) [reply .github/workflows/test.yaml]: Removed additional debug logs.

File: tests/credential/conftest.py (hunk 4 above)
Comment (Pushpendra-Garg) [reply tests/credential/conftest.py]: Modified year to 2026 for all files.

File: tests/credential/test_2200_async_create_cred.py (hunk 5 above)
Comment (Pushpendra-Garg) [reply tests/credential/test_2200_async_create_cred.py]: Done

File: tests/credential/test_2200_create_cred.py (hunk 6 above)
Comment (Pushpendra-Garg) [reply tests/credential/test_2200_create_cred.py]: Done.

File: tests/vector_index/test_5500_enable_disable_index.py (hunk 7 above)
Comment (Pushpendra-Garg) [reply tests/vector_index/test_5500_enable_disable_index.py]: Modified test by fetching with exact name and assert.

File: tests/vector_index/test_5400_list_index.py (hunk 10 above)
Comment (Pushpendra-Garg) [reply tests/vector_index/test_5400_list_index.py]: Yes, updated. The test now explicitly verifies that passing None to VectorIndex.list() does not fall back to the default match-all pattern ".*". Instead of only comparing counts, it checks that none of the managed test indexes are returned. The same clarification was applied to the async list test as well.

File: tests/credential/test_2200_create_cred.py (hunk 8 above)
Comment (Pushpendra-Garg) [reply tests/credential/test_2200_create_cred.py]: Fixed by replacing TEST_USER1 with a UUID-suffixed local user and keeping user cleanup in finally.

File: tests/provider/test_2400_async_enable.py (hunk 9 above)
Comment (Pushpendra-Garg) [reply tests/provider/test_2400_async_enable.py]: Fixed by using UUID-suffixed DB users instead of DB_USER1/DB_USER2 and dropping them in finally.

File: tests/vector_index/test_5400_list_index.py (hunk 11 above)
Comment (Pushpendra-Garg) [reply tests/vector_index/test_5400_list_index.py]: The test now verifies that passing an empty string to VectorIndex.list("") does not fall back to the default match-all pattern ".*" by checking that none of the managed test indexes are returned.

**Review by prateeks99 — DISMISSED** [review]

**Review by aosingh — APPROVED** [review]
